### PR TITLE
tests for message pkg

### PR DESCRIPTION
**Pull request by arkodg**
* specify len and capacity seperately in `make` api
* gen DeepCopy for infra ir structs

Signed-off-by: Arko Dasgupta <arko@tetrate.io>

### CODE REVIEW

File: internal/message/types_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gwapiv1b1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+func TestProviderResources(t *testing.T) {
```
Comment (danehans) [inline internal/message/types_test.go:14]: Can you add a test case where resources is nil?

xref: https://github.com/envoyproxy/gateway/pull/216

File: internal/message/types_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+			Name:      "test1",
+			Namespace: "test",
+		},
+	}
+	r1 := &gwapiv1b1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "route1",
+			Namespace: "test",
+		},
+	}
```
Comment (danehans) [inline internal/message/types_test.go:32]: Can you add a test case where gatewayclasses, gateways, and httproutes are nil?

xref: https://github.com/envoyproxy/gateway/pull/216

**Review by danehans — CHANGES_REQUESTED** [review]

**Review by danehans — APPROVED** [review]
